resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/SparkKubernetesClientFactory.scala

```diff
@@ -68,6 +69,8 @@ private[spark] object SparkKubernetesClientFactory extends Logging {
       .getOption(s"$kubernetesAuthConfPrefix.$CLIENT_KEY_FILE_CONF_SUFFIX")
     val clientCertFile = sparkConf
       .getOption(s"$kubernetesAuthConfPrefix.$CLIENT_CERT_FILE_CONF_SUFFIX")
+    // TODO(SPARK-37687): clean up direct usage of OkHttpClient, see also:
```

Comment:
Here are some backgroud on this note (also add the jira: [SPARK-37687](https://issues.apache.org/jira/browse/SPARK-37687)):

- There are [some problems (such as IPV6 based cluster support)](https://github.com/fabric8io/kubernetes-client/issues/2632) on okhttpclient v3, but it's a little bit diffcult to upgrade to v4 [1]
- Kubernetes client are also consider to[ support other clients](https://github.com/fabric8io/kubernetes-client/issues/3663#issuecomment-997402993) rather than single okhttpclient.
- Kubernetes client[ add a abstract layer](https://github.com/fabric8io/kubernetes-client/issues/3547) to support variety httpclient, okhttp client as one of supported http clients.

So, we better to consider to cleanup okhttpclient direct usage and use the httpclient which kubernetes client diret supported to reduce the potential risk in future upgrade.